tests/docker/test_container_manager.py

```diff
@@ -33,30 +35,21 @@ def test_start_stop(self):
         self.assertTrue(mock_client.remove_container.called)
 
     @gen_test
-    def test_containers_for_image_results(self):
-        ''' Test containers_for_image returns a list of Container '''
+    def test_containers_for_mapping_id(self):
+        ''' Test containers_for_mapping_id returns a list of Container '''
         # The mock client mocks the output of docker Client.containers
         docker_client = utils.mock_docker_client_with_running_containers()
         self.mock_docker_client = docker_client
         self.manager.docker_client.client = docker_client
 
-        # The output should be a list of Container
-        results = yield self.manager.containers_for_image("imageid")
-        expected = [Container(docker_id='someid',
-                              name='/remoteexec-image_3Alatest_user',
-                              image_name='simphony/mayavi-4.4.4:latest',  # noqa
-                              image_id='imageid', ip='0.0.0.0', port=None),
-                    Container(docker_id='someid',
-                              name='/remoteexec-image_3Alatest_user2',
-                              image_name='simphony/mayavi-4.4.4:latest',  # noqa
-                              image_id='imageid', ip='0.0.0.0', port=None),
```

Comment:
Please see comment below about `PublicPort`.  Here the `port` should not be `None`.
These three Containers capture scenarios when we have unsupported containers being started.  Removing them reduces coverage as well.